### ISSUE TITLE
msg.sender differs from test to system call

### ISSUE THREAD

**Issue by 0xhank** [latticexyz/mud#332]
I am getting odd behavior in my test suite -- my `msg.sender` value changes when I call a system, which breaks my test. This wasn't happening before I updated MUD to v31.1 -- any ideas why this is occurring?
```
testSpawn.t.sol:
    console.log("msg sender in spawn test:", msg.sender); // -> 0x1804c8
    console.log("player entity in spawn test:", playerEntity); -> 137122
    spawnSystem.executeTyped("Jamaican me crazy", Coord(1, 1));
```

``` 
execute() within SpawnSystem.sol:
    console.log("msg sender in spawn test:", msg.sender); // -> 0x416C4
    console.log("player entity in spawn test:", playerEntity); -> 373498
```

**Comment by alvrs** [comment]
Are you using foundry's [prank](https://book.getfoundry.sh/cheatcodes/prank) somewhere in `testSpawn.t.sol`? If not it's expected that the `msg.sender` in `execute` would be your test contract.

**Comment by 0xhank** [comment]
setting up a deployer prank has fixed the issue! Is this correct: the prank doesn't affect the `msg.sender` of the test function, just the `msg.sender` of all the functions it calls?

**Comment by alvrs** [comment]
That's correct - `prank` sets it only for the next function call, `startPrank` sets it for all following function calls until `stopPrank` is called
